crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -356,21 +356,19 @@ def upgrade_controller_nodes
         drbd_nodes = NodeObject.find("drbd_rsc:*")
         return upgrade_drbd_clusters unless drbd_nodes.empty?
 
-        founder = NodeObject.find(
-          "state:crowbar_upgrade AND pacemaker_founder:true"
-        ).first
+        founder = NodeObject.find("pacemaker_founder:true").first
         cluster_env = founder[:pacemaker][:config][:environment]
 
         non_founder = NodeObject.find(
-          "state:crowbar_upgrade AND pacemaker_founder:false AND " \
+          "pacemaker_founder:false AND " \
           "pacemaker_config_environment:#{cluster_env}"
         ).first
 
         return false unless upgrade_first_cluster_node founder.name, non_founder.name
```

Comment:
Sure, but that's only set after all step of the upgrade where completed. There are case where the node can be in ```ready``` but ```upgraded?``` still returns false. E.g. when ```disable_pre_upgrade_attribute_for``` failed in ```upgrade_next_cluster_node```